packages/sit-onyx/src/components/OnyxDataGrid/features/index.ts

```diff
@@ -117,13 +119,32 @@ export const useDataGridFeatures = <
       .filter((actions) => !!actions);
 
     return columns.map((column) => {
-      const iconActions = headerActions
+      const iconComponent = headerActions
         .flatMap((actionFactory) => actionFactory(column))
         .map(({ iconComponent }) => iconComponent);
 
+      const listItems = headerActions
+        .flatMap((actionFactory) => actionFactory(column))
+        .map(({ listItems }) => listItems);
+      let flyoutMenu: Component;
+
+      if (listItems.length > 0) {
+        flyoutMenu = h(
+          OnyxFlyoutMenu,
+          {
+            label: "",
+          },
+          {
+            button: iconComponent,
+            options: listItems,
+          },
+        );
+      }
+
       return {
         key: column,
-        component: () => h(HeaderCell, { label: String(column) }, { actions: iconActions }),
+        component: () =>
+          h(HeaderCell, { label: String(column) }, { actions: flyoutMenu ?? iconComponent }),
```

Comment:
```suggestion
          h(HeaderCell, { label: String(column) }, { actions: () => flyoutMenu ?? iconComponent }),
```

To get rid of console warnings, we should use getters for the slot content